File: app/src/main/java/com/elli0tt/rpg_life/presentation/screen/add_edit_characteristic/AddEditCharacteristicFragment.java
package com.elli0tt.rpg_life.presentation.screen.add_edit_characteristic;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.elli0tt.rpg_life.R;

public class AddEditCharacteristicFragment extends Fragment {

    private AddEditCharacteristicViewModel addEditCharacteristicViewModel;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_add_edit_characteristic, container, false);
    }
}

File: app/src/main/java/com/elli0tt/rpg_life/presentation/screen/add_edit_characteristic/AddEditCharacteristicFragment.kt
package com.elli0tt.rpg_life.presentation.screen.add_edit_characteristic

import android.os.Bundle
import android.view.View
import com.elli0tt.rpg_life.R
import com.elli0tt.rpg_life.presentation.core.fragment.BaseFragment
import com.elli0tt.rpg_life.presentation.extensions.injectViewModel
import com.elli0tt.rpg_life.presentation.screen.add_edit_characteristic.di.AddEditCharacteristicComponent

class AddEditCharacteristicFragment : BaseFragment(R.layout.fragment_add_edit_characteristic) {

    private lateinit var addEditCharacteristicComponent: AddEditCharacteristicComponent

    private lateinit var viewModel: AddEditCharacteristicViewModel

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        initDagger()
    }

    private fun initDagger() {
        addEditCharacteristicComponent =
            appComponent.addEditCharacteristicComponentFactory().create()
        addEditCharacteristicComponent.inject(this)

        viewModel = injectViewModel(viewModelFactory)
    }
}